src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1943,7 +1943,11 @@ bool isRefEscape
                         parameters[argsToParamsOpt.IsDefault ? argIndex : argsToParamsOpt[argIndex]] :
                         null;
 
-                    if (mixableArguments is not null && isMixableParameter(parameter))
+                    if (mixableArguments is not null
+                        && isMixableParameter(parameter)
+                        // assume any declaration variable is a valid mixing destination,
+                        // since we are going to infer a val-escape which makes it valid.
+                        && !IsLocalDeclarationExpression(argument, out _))
```

Comment:
>!IsLocalDeclarationExpression(argument, out _) [](http://example.com/codeflow?start=27&length=46)

Is it correct to skip an `out` declaration expression if the target parameter is `[UnscopedRef] out`?

Reply:
In this position, yes. Only val escape matters for mixing destinations.